### PR TITLE
[RDBMS] PostgreSQL restore API change update and Database CRUD

### PR DESCRIPTION
**Pull request by DaeunYim**
**Description**

- Restore API update: Latest API change is not updated to the Python SDK yet. After rolling out this version to prod worldwide, will update the Python SDK version. This will be done before code complete day
- Warning, help text updated: Fixing and modifying code based on CSS incidents - stop warning text, comprehensive help text for create command
- PG database CRUD update

**Testing Guide**

azdev test test_postgres_flexible_server_database_mgmt

**History Notes**

[RDBMS] Added PostgreSQL database create, show, and delete feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

**Comment by DaeunYim** [comment]
The CI test won't pass until we release the new Python SDK version (including restore source parameters change) after current train worldwide roll out.. Please review other code first (PG database CRUD),,,

**Comment by evelyn-ys** [comment]
#17191 has been merged, please sync with `dev` branch😊

**Comment by DaeunYim** [comment]
> #17191 has been merged, please sync with `dev` branch😊

Thank you!

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 1]
```diff
@@ -296,7 +296,7 @@ def _flexible_server_params(command_group):
                        arg_group='Authentication')
             c.argument('tags', tags_type)
             c.argument('public_access', options_list=['--public-access'],
-                       help='Determines the public access. Enter single or range of IP addresses to be included in the allowed list of IPs. IP address ranges must be dash-separated and not contain any spaces. Specifying 0.0.0.0 allows public access from any resources deployed within Azure to access your server. Specifying no IP address sets the server in public access mode but does not create a firewall rule. ',
+                       help='Determines the public access. Enter single or range of IP addresses to be included in the allowed list of IPs. IP address ranges must be dash-separated and not contain any spaces. Specifying 0.0.0.0 allows public access from any resources deployed within Azure to access your server. Setting it to "none" sets the server in public access mode but does not create a firewall rule. ',
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:299]:
```suggestion
                       help='Determines the public access. Enter single or range of IP addresses to be included in the allowed list of IPs. IP address ranges must be dash-separated and not contain any spaces. Specifying 0.0.0.0 allows public access from any resources deployed within Azure to access your server. Setting it to "None" sets the server in public access mode but does not create a firewall rule. ',
```

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 2]
```diff
@@ -317,6 +317,10 @@ def _flexible_server_params(command_group):
             if command_group == 'postgres':
                 c.argument('source_server', options_list=['--source-server'],
                            help='The name of the source server to restore from.')
+                c.argument('source_subscription_id', options_list=['--source-subscription-id'],
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:320]: `--source-subscription-id` is too long. Can you consider `--source-sub-id`?

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 3]
```diff
@@ -317,6 +317,10 @@ def _flexible_server_params(command_group):
...
+                           help='The subscription id of the source server.')
+                c.argument('source_resource_group_name', options_list=['--source-resource-group'],
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:322]: same here, considering a shorter name?

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 1 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:299]: Thanks for catching this! :)

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 2 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:320]: We decided to hide this parameters from customers and use same sub id and rg as the source server,, thanks!

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py [hunk 4]
```diff
@@ -566,24 +567,30 @@ def _create_database(db_context, cmd, resource_group_name, server_name, database
...
-        name=database_name,
-        charset=charset,
-        collation=collation
-    )
+    if charset is None and collation is None:
+        charset = 'utf8'
+        collation = 'utf8_general_ci'
+        logger.warning("Creating database with utf8 charset and utf8_general_ci collation")
+    elif charset or collation:
+        raise RequiredArgumentMissingError("charset and collation have to be iniput together.")
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:587]:
```suggestion
        raise RequiredArgumentMissingError("charset and collation have to be input together.")
```

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py [hunk 5]
```diff
@@ -142,17 +148,16 @@ def flexible_server_restore(cmd, client,
...
+    restore_point_in_time = change_str_to_datetime(restore_point_in_time)
     restore_point_in_time = restore_point_in_time.replace(tzinfo=dt.timezone.utc)
 
     parameters = postgresql_flexibleservers.models.Server(
         point_in_time_utc=restore_point_in_time,
         source_server_name=source_server,  # this should be the source server name, not id
         create_mode="PointInTimeRestore",
         availability_zone=zone,
+        source_resource_group_name=resource_group_name,
+        source_subscription_id=get_subscription_id(cmd.cli_ctx),
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py:160]: `get_subscription_id(cmd.cli_ctx)` is getting the login subscription. Is that possible that the `resource_group_name` or `source_server` belong to other sub?

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py (hunk 4 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:587]: Thanks Yishi!

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py (hunk 5 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py:160]: It won't be the case. It looks like there's complicate story from the backend team but we concluded that we keep the restore behavior as it was (no source server from different rg nor different sub)

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**